modules/openlmis-web/src/main/webapp/public/js/distribution/model/child-coverage.js

```diff
@@ -13,92 +13,141 @@ function ChildCoverage(facilityVisitId, childCoverageJSON) {
     var openedVialFieldsStatus = countNRStatus(this.openedVialLineItems, ['openedVial']);
     this.notRecordedApplied = (childCoverageFieldsStatus.notRecorded  +  openedVialFieldsStatus.notRecorded >
      childCoverageFieldsStatus.recorded + openedVialFieldsStatus.recorded);
   }
 
+  this.isOutdatedDistribution = checkIfIsOutdatedDistribution.call(this);
+  this.mandatoryFields = getMandatoryFields.call(this);
   init.call(this);
-}
 
-ChildCoverage.prototype.computeStatus = function (visited, review, ignoreSyncStatus) {
-  if (review && !ignoreSyncStatus) {
-    return DistributionStatus.SYNCED;
+  function getMandatoryFields() {
+    if(this.isOutdatedDistribution) {
+      return ['healthCenter11Months', 'outreach11Months', 'healthCenter23Months', 'outreach23Months'];
+    } else {
+      return ['maleHealthCenter11Months', 'femaleHealthCenter11Months',
+         'maleOutreach11Months', 'femaleOutreach11Months', 'maleHealthCenter23Months', 'femaleHealthCenter23Months',
+         'maleOutreach23Months', 'femaleOutreach23Months'];
+    }
   }
 
-  var status;
+  function isEmpty(obj) {
+    return (isUndefined(obj) || isUndefined(obj.value));
+  }
 
-  var isValid = function (field) {
-    if (!field)
-      return false;
-    return !(isUndefined(field.value) && !field.notRecorded);
-  };
+  function maleFemaleFieldsNotPresent(maleField, femaleField, totalField) {
+    return (isEmpty(maleField) && isEmpty(femaleField) && !isEmpty(totalField));
+  }
 
-  function validateLineItems(lineItems, mandatoryFields) {
-    $(lineItems).each(function (index, lineItem) {
-      $(mandatoryFields).each(function (index, field) {
-        if (lineItem.vaccination === 'Polio (Newborn)' && ['healthCenter23Months', 'outreach23Months'].indexOf(field) !== -1)
-          return true;
-        if (lineItem.vaccination === 'IPV' && ['healthCenter23Months', 'outreach23Months'].indexOf(field) !== -1)
-          return true;
-        if (lineItem.vaccination === 'Sarampo 2a dose' && ['healthCenter11Months', 'outreach11Months'].indexOf(field) !== -1)
-          return true;
-        if ((status === DistributionStatus.COMPLETE || !status) && isValid(lineItem[field])) {
-          status = DistributionStatus.COMPLETE;
-          return true;
-        } else if ((status === DistributionStatus.EMPTY || !status) && !isValid(lineItem[field])) {
-          status = DistributionStatus.EMPTY;
-          return true;
-        } else if ((status === DistributionStatus.EMPTY && isValid(lineItem[field])) || (status === DistributionStatus.COMPLETE && !isValid(lineItem[field]))) {
-          status = DistributionStatus.INCOMPLETE;
-          return false;
-        }
-        return true;
-      });
-      return status !== DistributionStatus.INCOMPLETE;
+  function checkIfIsOutdatedDistribution() {
+    outdated = false;
+    $(this.childCoverageLineItems).each(function (index, lineItem) {
```

Comment:
Does this function work as intended? It declares "outdated" within the global scope, and then sets it within its various conditionals. Based on how this methods is used (which is probably ideal), it seems as though the method should declare this variable with limited scope (via the "var" keyword) and then use statements like "return outdated;" rather than simply "return;"